app/models/invoice.rb

```diff
@@ -612,4 +615,15 @@ def notify_buyer_about_payment
       InvoiceMessenger.successfully_charged(self).deliver
     end
   end
+
+  private
+
+  # Allowed range for the invoice period is:
+  # - from: creation date of the buyer
+  # - to: 12 months from now
+  def period_range_valid?
+    return if self[:period]&.between?(buyer.created_at.to_date.beginning_of_month, Time.zone.now + 12.months)
```

Comment:
I decided to make it 12 months instead of 1 month from now, just to make it more flexible.

For example, we have an option to make friendly ID to include only the year, not the month, so maybe the provider wants to set the last month of the year as the invoice period, not the current month.

In general, I think it does not matter much how many months from the current date can be the maximum of the invoice period, the important thing is to have it limited, so that people do not put something like year 3800.

Reply:
OK, I think this logic actually has a problem...

Basically, it would prevent changes in valid, but old (more than 1 year ago) invoices...

I think we'll need to come up with something different.